concent_api/core/views.py

```diff
@@ -1220,6 +1225,632 @@ def store_message_and_message_status(
+            ack_report_computed_task    = ack_report_computed_task,
+            force_report_computed_task  = message.concents.ForceReportComputedTask(
+                report_computed_task = report_computed_task,
+            ),
+        )
+        mark_message_as_delivered_and_log(database_message, response_to_client)
+        return response_to_client
+
+    elif database_message.response_type == PendingResponse.ResponseType.ForceGetTaskResultRejected.name:  # pylint: disable=no-member
+        report_computed_task = deserialize_message(database_message.subtask.report_computed_task.data.tobytes())
+        response_to_client = message.concents.ForceGetTaskResultRejected(
+            force_get_task_result = message.concents.ForceGetTaskResult(
+                report_computed_task = report_computed_task,
+            )
+        )
+        mark_message_as_delivered_and_log(database_message, response_to_client)
+        return response_to_client
+
+    elif database_message.response_type == PendingResponse.ResponseType.ForceGetTaskResultFailed.name:  # pylint: disable=no-member
+        task_to_compute = deserialize_message(database_message.subtask.task_to_compute.data.tobytes())
+        response_to_client = message.concents.ForceGetTaskResultFailed(
+            task_to_compute = task_to_compute,
+        )
+        mark_message_as_delivered_and_log(database_message, response_to_client)
+        return response_to_client
+
+    elif (
+        database_message.response_type in [
+            PendingResponse.ResponseType.ForceGetTaskResultUpload.name,  # pylint: disable=no-member
+            PendingResponse.ResponseType.ForceGetTaskResult.name,  # pylint: disable=no-member
+        ]
+    ):
+        report_computed_task    = deserialize_message(database_message.subtask.report_computed_task.data.tobytes())
+        file_transfer_token     = create_file_transfer_token(
+            report_computed_task,
+            client_public_key,
+        )
+        # assert client_public_key is database_message.subtask.provider.public_key_bytes
+
+        file_transfer_token.operation = 'upload'
+
+        response_to_client = message.concents.ForceGetTaskResultUpload(
+            file_transfer_token     = file_transfer_token,
+            force_get_task_result   = message.concents.ForceGetTaskResult(
+                report_computed_task = report_computed_task,
+            )
+        )
+        mark_message_as_delivered_and_log(database_message, response_to_client)
+        return response_to_client
+
+    elif database_message.response_type == PendingResponse.ResponseType.ForceGetTaskResultDownload.name:  # pylint: disable=no-member
+        report_computed_task    = deserialize_message(database_message.subtask.report_computed_task.data.tobytes())
+        file_transfer_token     = create_file_transfer_token(
+            report_computed_task,
+            client_public_key,
+        )
+        file_transfer_token.operation = 'download'
```

Comment:
Why not make `operation` a parameter of `create_file_transfer_token()`?

Reply:
Done